src/ImageSharp.Web.Providers.Azure/Providers/AzureBlobStorageImageProvider.cs

```diff
@@ -94,19 +71,24 @@ public class AzureBlobStorageImageProvider : IImageProvider
         /// <inheritdoc/>
         public async Task<IImageResolver> GetAsync(HttpContext context)
         {
-            // Strip the leading slash and container name from the HTTP request path and treat
-            // the remaining path string as the blob name.
-            // Path has already been correctly parsed before here.
-            string blobName = context.Request.Path.Value.TrimStart(SlashChars)
-                                     .Substring(this.storageOptions.ContainerName.Length)
-                                     .TrimStart(SlashChars);
-
-            if (string.IsNullOrWhiteSpace(blobName))
+            var parts = new AzureBlobStoragePathParts(context.Request.Path.Value, this.storageOptions.RoutePrefix);
+
+            if (string.IsNullOrEmpty(parts.BlobFilename))
+            {
+                return null;
+            }
+
+            CloudBlobClient client = this.storageAccount.CreateCloudBlobClient();
```

Comment:
Create the client and containers at startup. The options should take a collection of strings representing the containers to register.